pkg/types/validation/installconfig.go

```diff
@@ -115,6 +115,10 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	}
 	allErrs = append(allErrs, validateCloudCredentialsMode(c.CredentialsMode, field.NewPath("credentialsMode"), c.Platform.Name())...)
 
+	if c.Publish == types.InternalPublishingStrategy && c.Platform.AWS == nil && c.Platform.Azure == nil && c.Platform.GCP == nil {
```

Comment:
Since the list of public clouds will grow over time, I wonder if there's a better way to group them instead have to add `&& c.Platform.Cloud == nil` for each platform.